src/tests/ftest/util/agent_utils.py

```diff
@@ -144,59 +110,59 @@ def run_agent(basepath, server_list, client_list=None):
                 output = sessions[client].stderr.read()
             except IOError as excpn:
                 if excpn.errno != errno.EAGAIN:
-                    raise AgentFailed("Error in starting daos_agent: "
-                                      "{0}".format(str(excpn)))
+                    raise AgentFailed(
+                        "Error in starting daos_agent: {0}".format(str(excpn)))
                 time.sleep(1)
                 continue
             expected_data += output
 
             match = re.findall(pattern, output)
             if len(match) > 0:
-                print("<AGENT> agent started on node {} in {} "
-                      "seconds".format(client, time.time() - start_time))
+                print(
+                    "<AGENT> agent started on node {} in {} seconds".format(
+                        client, time.time() - start_time))
                 break
 
     return sessions
-# pylint: enable=too-many-locals
 
 
 def stop_agent(sessions, client_list=None):
-    """
-    This should kill ssh and the agent.  This is temporary; presuming the
-    agent will deamonize at somepoint and can be started killed more
-    appropriately.
+    """Kill ssh and the agent.
 
-    client_list -- kill daos_agent on these hosts
-    sessions    -- set of subprocess sessions returned by run_agent()
-    """
+    This is temporary; presuming the agent will deamonize at somepoint and can
+    be started killed more appropriately.
+
+    Args:
+        sessions (dict): set of subprocess sessions returned by run_agent()
+        client_list (list, optional): lists of hosts running the daos agent.
+            Defaults to None.
+
+    Raises:
+        AgentFailed: if the daos agents failed to stop
 
+    """
     # if empty client list, 'self' is effectively client
     if client_list is None:
         client_list = [socket.gethostname().split('.', 1)[0]]
 
-    # this kills the agent
-    for client in client_list:
-        cmd = "pkill daos_agent"
-        resp = subprocess.call(["ssh", client, cmd])
+    # Kill the agents processes
+    clush(client_list, "pkill daos_agent", None, False)
 
+    # Kill any processes running in the sessions
     for client in sessions:
         if sessions[client].poll() is None:
             sessions[client].kill()
         sessions[client].wait()
 
-    # check to make sure it's dead
+    # Check to make sure all the daos agents are dead
+    # pgrep exit status:
+    #   0 - One or more processes matched the criteria.
+    #   1 - No processes matched.
+    #   2 - Syntax error in the command line.
+    #   3 - Fatal error: out of memory etc.
     time.sleep(5)
```

Comment:
Not a fan of this kind of arbitrary sleep (and let's hope we slept long enough).  Can we replace this with a query/wait loop of what we are waiting for?  Keep in mind that looping locally means lots of expensive `ssh`es vs. looping remotely is done in a single ssh.

Reply:
I completely agree about arbitrary sleeps.  It would be great to know why it was added in the first place.  Given that the `Popen.poll()` checks to see if the child process (daos_agent) has terminated and the `Popen.wait()` waits for the child process to terminate (which should probably be indented with the `Popen.kill()`), I don't see an obvious need for the sleep.  I assume at some point there was a need and removing it could cause potential CI issues even if they are not seen during this PR.  Should the removal of sleeps be a separate PR?

Reply:
Frankly I suspect it was added "just in case" and that we should remove the for all of the reasons you stated.

I suppose that could be a different PR though.

Reply:
I think this will be handled by the changes going on in #911

Reply:
I don't see that patch even touching `src/tests/ftest/util/agent_utils.py`.

At the same time, still not saying that this needs to be resolved in this PR.